doc/userguide/rules/enip-keyword.rst

```diff
@@ -1,40 +1,225 @@
 ENIP/CIP Keywords
 =================
 
-The enip_command and cip_service keywords can be used for matching on various properties of
-ENIP requests.
+enip_command
+------------
 
-There are three ways of using this keyword:
+For the ENIP command, we are matching against the command field found in the ENIP encapsulation.
+
+Examples::
 
-* matching on ENIP command with the setting "enip_command";
-* matching on CIP Service with the setting "cip_service".
-* matching both the ENIP command and the CIP Service with "enip_command" and "cip_service" together
+  enip_command:99;
+  enip_command:ListIdentity;
 
 
-For the ENIP command, we are matching against the command field found in the ENIP encapsulation.
+cip_service
+-----------
 
 For the CIP Service, we use a maximum of 3 comma separated values representing the Service, Class and Attribute.
 These values are described in the CIP specification. CIP Classes are associated with their Service, and CIP Attributes
 are associated with their Service. If you only need to match up until the Service, then only provide the Service value.
 If you want to match to the CIP Attribute, then you must provide all 3 values.
 
-
-Syntax::
-
-  enip_command:<value>
-  cip_service:<value(s)>
-  enip_command:<value>, cip_service:<value(s)>
-
-
 Examples::
 
-  enip_command:99
   cip_service:75
   cip_service:16,246,6
-  enip_command:111, cip_service:5
 
 
 (cf. http://read.pudn.com/downloads166/ebook/763211/EIP-CIP-V1-1.0.pdf)
 
 Information on the protocol can be found here:
 `<http://literature.rockwellautomation.com/idc/groups/literature/documents/wp/enet-wp001_-en-p.pdf>`_
+
+enip.status
+-----------
+
+For the ENIP status, we are matching against the status field found in the ENIP encapsulation.
+It uses a 32-bit unsigned integer as value.
+
+Examples::
+
+  enip.status:100;
+  enip.status:>106;
+
+enip.protocol_version
+---------------------
+
+Match on the protocol version in different messages.
+It uses a 16-bit unsigned integer as value.
+
+Examples::
+
+  enip.protocol_version:1;
+  enip.protocol_version:>1;
+
+enip.cip_attribute
+------------------
+
+Match on the cip attribute in different messages.
+It uses a 32-bit unsigned integer as value.
+
+This allows to match without needing to match on cip.service.
+
+Examples::
+
+  enip.cip_attribute:1;
+  enip.cip_attribute:>1;
+
+enip.cip_instance
+-----------------
+
+Match on the cip instance in CIP request path.
+It uses a 32-bit unsigned integer as value.
+
+Examples::
+
+  enip.cip_instance:1;
+  enip.cip_instance:>1;
+
+enip.cip_class
+--------------
+
+Match on the cip class in CIP request path.
+It uses a 32-bit unsigned integer as value.
+
+This allows to match without needing to match on cip.service.
+
+Examples::
+
+  enip.cip_class:1;
+  enip.cip_class:>1;
+
+enip.cip_extendedstatus
+-----------------------
+
+Match on the cip extended status if any (one of them in case of multiple service packet).
```

Comment:
This sentence is a bit confusing to me. Does this mean that `cip_extendedstatus` will match on extended status if there is any, and that it can match on any of the seen statuses - as is, a single packet might have more than one?

Reply:
Your understanding is correct.

Do you have another way to phrase this ?

Reply:
What about...
Match on the cip extended status, if any is present. For multiple service packet, if there is more than one, can match on any of the seen statuses.